app/utils/injectReducer.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './reducerInjectors';
+
+/**
+ * Dynamically injects a reducer
+ *
+ * @param {string} name Name of the reducer
+ * @param {function} reducer A reducer that will be injected
+ * @param {string} [mode] By default ('restart-on-remount') the reducer will be started on component mount and removed
+ * on component un-mount for improved performance. Another option is 'daemon' that starts the reducer on component mount and
+ * never removes it.
+ */
+export default ({ name, reducer, mode }) => (WrappedComponent) => {
+  class ReducerInjector extends React.Component {
+    static WrappedComponent = WrappedComponent;
+    static contextTypes = {
+      store: PropTypes.object.isRequired,
+    };
+    static displayName = `withReducer(${(WrappedComponent.displayName || WrappedComponent.name || 'Component')})`;
+
+    componentWillMount() {
+      const { injectReducer } = this.injectors;
+
+      injectReducer(name, reducer);
+    }
+
+    componentWillUnmount() {
+      const { ejectReducer } = this.injectors;
```

Comment:
unlike sagas, reducers are pure and don't create any side effects.. i have pruned state tree but not personally had the need to hook reducers into component lifecycle... i feel like it could also cause some unexpected behavior, since state and actions aren't always contained to single component lifecycle, like ui events in a different page or background events like incoming websockets. obviously these should all really be root sagas, but this not may be immediately obvious to new users